java/arcs/core/host/EntityHandleManager.kt

```diff
@@ -56,23 +110,62 @@ class EntityHandleManager(
      */
     suspend fun <T : Entity> createSingletonHandle(
         mode: HandleMode,
-        name: String,
+        baseName: String,
         entitySpec: EntitySpec<T>,
         storageKey: StorageKey,
         idGenerator: Id.Generator = Id.Generator.newSession()
-    ) = handleManager.rawEntitySingletonHandle(
-        storageKey,
-        entitySpec.SCHEMA,
-        name = idGenerator.newChildId(
+    ): BaseHandleAdapter {
+        val name = idGenerator.newChildId(
             idGenerator.newChildId(arcId.toArcId(), hostId),
-            name
+            baseName
         ).toString()
-    ).let {
-        when (mode) {
-            HandleMode.Read -> ReadSingletonHandleAdapter(entitySpec, it)
-            HandleMode.Write -> WriteSingletonHandleAdapter<T>(it, idGenerator)
+
+        val entityPrep = EntityPrep<T>(
+            name,
+            idGenerator,
+            entitySpec.SCHEMA,
+            Ttl.Infinite,
```

Comment:
ttl should be an arg for createSingletonHandle and createCollectionHandle (you can also just leave a todo for me to plumb it through)

Reply:
Yes, I planned for that to be addressed in a subsequent commit.